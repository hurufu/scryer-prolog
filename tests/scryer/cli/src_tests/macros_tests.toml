args = [
    "-f",
    "--no-add-history",
    "-g", "use_module(library(format)),listing(example/1)",
    "-g", "halt",
    "src/tests/macros.pl"
]
