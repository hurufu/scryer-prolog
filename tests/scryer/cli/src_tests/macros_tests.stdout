example(formated_string(A)) :-
   A="This is my term foo(1)".
example(bignum_example(A,B,C)) :-
   (  A<B,
      !,
      A>0
   ;  B+1<C
   ),
   compute(A,B).
example(numeric_enum_example) :-
   foo(100,101),
   bar([110,111|A]).
example(ascii_examples) :-
   (  foo([0]) ->
      foo([7])
   ;  foo([8])
   ).
example(base_conversion) :-
   foo([171,205,239,1,35,69,96]).
example(quotated_goals_are_not_expanded(A)) :-
   quote#bignum#(A<2).
example(quotation_works_in_nested_terms(A)) :-
   A=[111,B,quote#fep#putb].
example(expands_arithmetic_functions(A,B)) :-
   B is A+113/24.
example(doesnt_expand_uninstantiated_macros(A)) :-
   B=fep#A.
example(compilation([A,B,C])) :-
   A=24,
   B=36,
   C=72.
